docs/what-is-ck.rst

```diff
@@ -0,0 +1,40 @@
+.. meta::
+  :description: Composable Kernel documentation and API reference library
+  :keywords: composable kernel, CK, ROCm, API, documentation
+
+.. _what-is-ck:
+
+********************************************************************
+What is the Composable Kernel Library
+********************************************************************
+
+-----------
+Methodology
+-----------
+
+The Composable Kernel (CK) library provides a programming model for writing performance critical kernels for machine learning workloads across multiple architectures including GPUs and CPUs, through general purpose kernel languages like HIP C++.
+
+CK utilizes two concepts to achieve performance portability and code maintainability:
+
+* A tile-based programming model
+* Algorithm complexity reduction for complex ML operators using an innovative technique called
+  "Tensor Coordinate Transformation".
+
+.. image:: data/ck_component.png
+   :alt: CK Components
+
+--------------
+Code Structure
+--------------
+
+The CK library is structured into 4 layers:
+
+* "Templated Tile Operators" layer
+* "Templated Kernel and Invoker" layer
+* "Instantiated Kernel and Invoker" layer
+* "Wrapper for tensor transform operations"
+* "Client API" layer
```

Comment:
It's a bit vague here - we say there are 4 layers but list 5 components.

Maybe we could change it to sth like that? @bartekxk, I guess you added it. Does it make sense?
```suggestion
The CK library is structured into 4 layers:

* "Templated Tile Operators" layer
* "Templated Kernel and Invoker" layer
* "Instantiated Kernel and Invoker" layer
* "Client API" layer

It also includes a simple wrapper component used to perform tensor transform operations more easily and with fewer lines of code.
```

Reply:
Hi Bartek, I also noticed the additional bullet, but I did not make this change. I can change it as you requested, but it is from an earlier PR than mine. Let me know if you think I should implement your suggestions. Thanks.